bundles/org.openhab.binding.heliosventilation/src/main/java/org/openhab/binding/heliosventilation/internal/HeliosVentilationHandler.java

```diff
@@ -0,0 +1,469 @@
+                try {
+                    try {
+                        // wait a little to increase the likelihood that the complete frame is already buffered.
+                        // this improves the robustness for RS485/USB converters which sometimes duplicate bytes
+                        // otherwise
+                        Thread.sleep(5);
+                    } catch (InterruptedException e) {
+                        // ignore interruption
+                    }
+                    byte frame[] = { 0, 0, 0, 0, 0, 0 };
+                    do {
+                        int cnt = 0;
+                        int c;
+
+                        do {
+                            c = inputStream.read();
+                            if (cnt > 0 || c == 0x01) {
+                                frame[cnt] = (byte) c;
+                                cnt++;
+                            }
+
+                            if (cnt < 6 && inputStream.available() < 1) {
+                                // frame not yet complete but no input available, let's wait a little to merge
+                                // interrupted transmissions
+                                try {
+                                    Thread.sleep(10);
+                                } catch (InterruptedException e) {
+                                    // ignore interruption
+                                }
+                            }
+                        } while (inputStream.available() > 0 && (c != -1) && cnt < 6);
+
+                        int sum = checksum(frame);
+                        if (sum == (frame[5] & 0xff)) {
+                            logger.trace("HeliosVentilation: Read from serial port: {}",
+                                    String.format("%02x %02x %02x %02x", frame[1], frame[2], frame[3], frame[4]));
+                            interpretFrame(frame);
+
+                        } else {
+                            logger.trace(
+                                    "HeliosVentilation: Read frame with not matching checksum from serial port: {}",
+                                    String.format("%02x %02x %02x %02x %02x %02x (expected %02x)", frame[0], frame[1],
+                                            frame[2], frame[3], frame[4], frame[5], sum));
+
+                        }
+
+                    } while (inputStream.available() > 0);
+
+                } catch (IOException e1) {
+                    logger.debug("Error reading from serial port: {}", e1.getMessage(), e1);
+                }
+                break;
+            default:
+                break;
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        disconnect();
+        super.handleRemoval();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!isConnected()) {
+            connect(); // let's try to reconnect if the connection failed or was never established before
```

Comment:
`handleCommand` is (like `initialize`) expected to return fast. In `initialize` you schedule `connect` instead of executing it directly. The same is true for handling of `REFRESH`, polling might take some time. Probably it would  be better  to  have something like `handleCommandInternal` and schedule that one for later execuion outside of the main thread.